### DISCUSSION TITLE
Is BroadcastSystemMessage necessary ?

### DISCUSSION THREAD

**Discussion by uhonermann** [dokan-dev/dokany#1186] [Q&A] [answered]
Hello,

the call to BroadcastSystemMessage after mounting the drive takes between 5 and 30 seconds on some systems.
So I wonder if it is really necessary to call it.

Regards,
   Uwe

**Comment by Liryna** [comment]
The mount is already broadcast by the library on mount and unmount. https://github.com/dokan-dev/dokany/blob/036fc023342050054c926030804b01f5b2c1b975/dokan/mount.c#L508 Maybe a third app is delaying its effect ?

Could it be due to https://github.com/dokan-dev/dokany/issues/528 ?

**Comment by uhonermann** [comment]
Hi Liryna,

it's exactly the call to BroadcastSystemMessage from DokanBroadcastLink, that is taking such a large amount of time (on some, but unfortunately many, machines).
I think, it is because of some applications that delay it, but it seems to be hard to find these applications (already tried using Spy++).

So I was wondering what is going to happen if I simply remove the BroadcastSystemMessage call.

Regards,
   Uwe

**Comment by Liryna** [comment, marked as answer]
If we don't call `BroadcastSystemMessage`, explorer or other app are not going to reflect the mount / unmount in their UI or else.

It is sadly "expected" that `BroadcastSystemMessage` hangs due to external apps. What we should do here is starting a thread and execute this call separately to the mount thread.

**Comment by uhonermann** [comment]
Hm, yes, you're absolutely right, that would be the perfect solution.

**Comment by uhonermann** [comment]
I just tried it completely without the BroadcastSystemMessage, and couldn't find any problem so far.
The mount is extremely fast now (less than a second).

Okay, we mount a network drive for current session, maybe it makes a difference if mounting a local disk device globally.
But we use local disk device only for server systems, where the speed of mounting doesn't really matter.

**Reply by Liryna** [reply]
I have created https://github.com/dokan-dev/dokany/issues/1187 to track the change. Thanks!

**Reply by Liryna** [reply]
This is done with https://github.com/dokan-dev/dokany/commit/d715188884ece779d903f08b845c6510c96146c6

**Comment by uhonermann** [comment]
Hi Liryna,

our customer confirms, that the delay does not happen anymore with that version.

Regards,
  Uwe

**Reply by Liryna** [reply]
Thanks for the confirmation!
